packages/expo-video-thumbnails/ios/VideoThumbnailsModule.swift

```diff
@@ -21,10 +21,16 @@ public class VideoThumbnailsModule: Module {
     let generator = AVAssetImageGenerator.init(asset: asset)
 
     generator.appliesPreferredTrackTransform = true
-    generator.requestedTimeToleranceBefore = CMTime.zero
     generator.requestedTimeToleranceAfter = CMTime.zero
 
     let time = CMTimeMake(value: options.time, timescale: 1000)
+
+    // `requestedTimeToleranceBefore` can only be set if `time` is less
+    // than the video duration, otherwise it will fail to generate an image.
+    if(time < asset.duration) {
+      generator.requestedTimeToleranceBefore = CMTime.zero
+    }
```

Comment:
```suggestion
    if time < asset.duration {
      generator.requestedTimeToleranceBefore = .zero
    }
```

Reply:
Are you sure that .zero is valid?

Reply:
Yeah, that should be fine. Are you seeing an error? 🤔